docs/guides/server/bootstrap-admin-recovery.adoc

```diff
@@ -0,0 +1,90 @@
+
+<@tmpl.guide
+title="Admin bootstrap and recovery"
+summary="Learn how to bootstrap and recover admin account." >
+
+== A solution for a disaster scenario
+
+It used to be difficult to regain access to a {project_name} instance, where all admin users have been locked out. The process required multiple user-advanced steps, including direct database access and manual changes. In the continuous effort to improve the user experience, {project_name} now provides multiple ways to bootstrap a new admin account, which can be later used to recover from such a situation.
+
+== Bootstrapping an admin user using a dedicated command
+
+To create a temporary admin user, execute the following command:
+
+<@kc.bootstrapadmin parameters="user"/>
+
+This command can be executed even before the {project_name} first ever start. In other words, it does not matter, whether the `master` realm already exists or not. If no other parameters are specified and/or no corresponding environment variables are set, a user is prompted to enter required information. For more information, see the <<Environment variables>> section bellow.
+
+Alternatively, the parameters can be directly specified in the command:
+
+<@kc.bootstrapadmin parameters="user --username tmpadm --password tmpadmpass"/>
+
+This command creates a temporary admin user with the username `tmpadm` and the password `tmpadmpass`. The user is valid for 60 minutes.
+
+== Bootstrapping an admin service account
+
+In automated scenarios, an admin service account can be more suitable alternative to a temporary admin user.
+
+Again, to create a temporary admin service account, execute the following command:
+
+<@kc.bootstrapadmin parameters="service"/>
+
+The same rule applies here - the command can be executed regardless of the `master` realm existence. If no corresponding environment variables nor the additional parameter are set, a user will be prompted to enter the required information. For more information, see the <<Environment variables>> section bellow.
+
+Alternatively, you can specify the parameters directly in the command:
+
+<@kc.bootstrapadmin parameters="service --client-id tmpclient --client-secret:env=SECRET_VAR"/>
+
+This command creates a temporary admin service account with the client ID `tmpclient` and the secret retrieved from the environment variable.
+
+== Regaining access to the realm with an increased security
+
+Passwordless, OTP or other advanced authentication methods can be enforced for the realm with lost admin access. In such case, the admin service account needs to be created. After the service account is created, authentication against the {project_name} instance is required to perform all the necessary operations:
+
+[source]
+----
+POST realms/{realm}/protocol/openid-connect/token \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -d "client_id=<service_account_id>" \
+  -d "client_secret=<service_account_secret>" \
+  -d "grant_type=client_credentials"
```

Comment:
We could probably use kcadm.sh|bat in these examples right away.

Reply:
Adressed.